### PR TITLE
Remove brackets feature option

### LINKED ISSUE #892: Potential KeyValueProcessor feature: include_brackets
A boolean specifying whether to treat square brackets, angle brackets, and parentheses as value "wrappers" that should be removed from the value.

See: https://www.elastic.co/guide/en/logstash/current/plugins-filters-kv.html#plugins-filters-kv-include_brackets

### PR DESCRIPTION
**Pull request by shenkw1**
### Description
Specify whether to keep or remove square brackets, curly braces, and/or parentheses
 
### Issues Resolved
#892 
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java [hunk 1]
```diff
@@ -195,6 +201,10 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     key = transformKey(key);
                 }
 
+                if (keyValueProcessorConfig.getRemoveBrackets()) {
+                    value = value.toString().replaceAll("[\\[\\]()<>]","");
```
Comment (chenqi0805) [inline data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java:205]: Use constant variable to indicate the pattern meaning

File: data-prepper-plugins/key-value-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessorTests.java [hunk 2]
```diff
@@ -466,6 +467,20 @@ void testTrueThreeInputsDuplicateValuesKvProcessor() {
         assertThatKeyEquals(parsed_message, "key1", expectedValue);
     }
 
+    @Test
+    void testTrueRemoveBracketsKvProcessor() {
+        when(mockConfig.getRemoveBrackets()).thenReturn(true);
+
+        final Record<Event> record = getMessage("key1=(value1)&key2=[value2]&key3=<value3>");
```
Comment (chenqi0805) [inline data-prepper-plugins/key-value-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessorTests.java:474]: Shall we add some more corner cases? e.g. what is the expected output for the following?

`key1=((value1)`, `key1=[value1][value2]`

File: data-prepper-plugins/key-value-processor/README.md [hunk 3]
```diff
@@ -78,6 +78,7 @@ When run, the processor will parse the message into the following output:
   * Default: `false`
   * Example: `remove_brackets` is `true`. `{"key1=(value1)"}` will parse into `{"key1": value1}`
   * Example: `remove_brackets` is `false`. `{"key1=(value1)"}` will parse into `{"key1": "(value1)"}`
+  * In the case of a key-value pair with a brackets and a split character, the behavior when `remove_brackets` is `true` will look like the following: `{key1=(value1&value2)}` will parse into `{"key1":"value1","value2)":null}`
```
Comment (chenqi0805) [inline data-prepper-plugins/key-value-processor/README.md:81]: To rephrase it more accurately, the splitting will take priority over `remove_brackets=true`, e.g. `{key1=(value1&value2)}` will parse into `{"key1":"value1","value2)":null}` instead of `{"key1":"value1","value2":null}`

**Review by chenqi0805 — APPROVED** [review]

**Review by asifsmohammed — APPROVED** [review]
